### ISSUE TITLE
`ignore null` causes parsing error for Redshift

### ISSUE THREAD

**Issue by puetsch** [sqlfluff/sqlfluff#1629]
Ran `sqfluff lint` on query containing the code below

```sql
first_value(finalsaleprice ignore nulls)
```

## Expected Behaviour
Passes as valid SQL
(see https://docs.aws.amazon.com/redshift/latest/dg/r_WF_first_value.html)

## Observed Behaviour
"Found unparsable section: ' ignore nulls'"

## Steps to Reproduce
Run `sqfluff lint` on query containing the code above

## Dialect
postgres

## Version
0.6.5
Python 3.8.11

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = postgres
templater = jinja
rules = None
exclude_rules = L011,L031,L034
recurse = 0
output_line_length = 80
runaway_limit = 10
ignore_templated_areas = True
encoding = autodetect
# Comma separated list of file extensions to lint.

# NB: This config will only apply in the root folder.
sql_file_exts = .sql,.sql.j2,.dml,.ddl

[sqlfluff:indentation]
indented_joins = False
indented_using_on = True
template_blocks_indent = True

[sqlfluff:templater]
unwrap_wrapped_queries = True

[sqlfluff:templater:jinja]
apply_dbt_builtins = True

[sqlfluff:templater:jinja:macros]
# Macros provided as builtins for dbt projects
dbt_ref = {% macro ref(model_ref) %}{{model_ref}}{% endmacro %}
dbt_source = {% macro source(source_name, table) %}{{source_name}}_{{table}}{% endmacro %}
dbt_config = {% macro config() %}{% for k in kwargs %}{% endfor %}{% endmacro %}
dbt_var = {% macro var(variable) %}item{% endmacro %}
dbt_is_incremental = {% macro is_incremental() %}True{% endmacro %}

# Some rules can be configured directly from the config common to other rules.
[sqlfluff:rules]
tab_space_size = 4
max_line_length = 120
indent_unit = space
comma_style = trailing
allow_scalar = True
single_table_references = consistent
unquoted_identifiers_policy = all

# Some rules have their own specific config.
[sqlfluff:rules:L003]
lint_templated_tokens = True

[sqlfluff:rules:L007]  # Keywords
operator_new_lines = after

[sqlfluff:rules:L010]  # Keywords
capitalisation_policy = consistent

[sqlfluff:rules:L014]  # Unquoted identifiers
extended_capitalisation_policy = consistent

[sqlfluff:rules:L016]
ignore_comment_lines = False

[sqlfluff:rules:L029]  # Keyword identifiers
unquoted_identifiers_policy = aliases

[sqlfluff:rules:L030]  # Function names
capitalisation_policy = consistent

[sqlfluff:rules:L038]
select_clause_trailing_comma = forbid

[sqlfluff:rules:L040]  # Null & Boolean Literals
capitalisation_policy = consistent

[sqlfluff:rules:L042]
# By default, allow subqueries in from clauses, but not join clauses.
forbid_subquery_in = join

[sqlfluff:rules:L047]  # Consistent syntax to count all rows
prefer_count_1 = False
prefer_count_0 = False

```

**Comment by tunetheweb** [comment]
Could you please fill in the bug templated when raising issues. There's not enough info in that.

Which dialect are you using because Postgres is not one of the supported dialects.

What is your SQL, because that is a snippet of SQL and not a valid SQL statement.

What version of SQL Fluff are you using?

**Comment by puetsch** [comment]
That's correct. Technically including an `over()`, so
```sql
select first_value(finalsaleprice ignore nulls) over () as c1
from table1
```

**Comment by puetsch** [comment]
According to the Redshift documentation it's not a valid syntax but, you're right, it parses.

In the [Postgres Doc](https://www.postgresql.org/docs/9.4/functions-window.html), it states
> The SQL standard defines a RESPECT NULLS or IGNORE NULLS option for lead, lag, first_value, last_value, and nth_value. This is not implemented in PostgreSQL

According to [ANSI SQL standard,](https://docs.oracle.com/cd/B19306_01/server.102/b14200/functions057.htm) the `RESPECT NULLS` would have to be inside the parentheses.

**Comment by puetsch** [comment]
Thank you, appreciate it! I tried it out today but got a few other errors where postgres worked better. I'll dig in a bit more an open issue reports.
